tests/feature_elimination/test_feature_elimination.py

```diff
@@ -295,6 +295,33 @@ def test_calculate_number_of_features_to_remove():
     )
```

Comment:
I've modified your unit test to be independent from the other unit tests (X, y, weight) such that when we refactor other unit tests it is easier to work with:

```
def test_shap_automatic_num_feature_selection():
    """
    Test automatic num feature selection methods
    """
    X = pd.DataFrame(
        {
            "col_1": [1, 0, 1, 0, 1, 0, 1, 0],
            "col_2": [0, 0, 0, 0, 0, 1, 1, 1],
            "col_3": [1, 1, 1, 0, 0, 0, 0, 0],
        }
    )
    y = pd.Series([0, 0, 0, 0, 1, 1, 1, 1])
    
    clf = DecisionTreeClassifier(max_depth=1, random_state=1)
    shap_elimination = ShapRFECV(
        clf,
        random_state=1,
        step=1,
        cv=2,
        scoring="roc_auc",
        n_jobs=1,
    )
    _ = shap_elimination.fit_compute(X, y, approximate=True, check_additivity=False)
    
    best_features = shap_elimination.get_reduced_features_set(num_features="best")
    best_coherent_features = shap_elimination.get_reduced_features_set(
        num_features="best_coherent",
    )
    best_parsimonious_features = shap_elimination.get_reduced_features_set(
        num_features="best_parsimonious"
    )
    
    assert best_features == ["col_3"]
    assert best_coherent_features == ["col_1", "col_2", "col_3"]
    assert best_parsimonious_features == ["col_3"]
```

It should also simplify the test a bit I hope.

Reply:
TY!